### DISCUSSION TITLE
Partition FieldSets?

### DISCUSSION THREAD

**Discussion by hblumberg** [Parcels-code/Parcels#1304] [Q&A] [answered]
Is it possible to partition a `FieldSet` into smaller `FieldSet`s?

The motivation is that we want to create a `FieldSet` where the scaling factor varies by time. For example, if we have a `FieldSet` with two days of data, we might want to apply a scaling factor of 0.5 to the first day and a factor of 0.75 to the second day. So we were wondering if there is a way to take the original `FieldSet` and partition it (either by a particular time or into equal chunks).

Currently, we're thinking our best bet is to load the data into an `xarray`, partition it there, and then create `FieldSet`s from the partitions. Before we implement that, we wanted to check to see if anyone here has better ideas!

Thanks in advance!

**Comment by erikvansebille** [comment, marked as answer]
Hi @hblumberg, thanks for your question. I don't think that what you want to do is possible out-of-the-box with `FieldSet`s. 

Options to still implement this are
1. as you suggest yourself, by using array, as you suggest. array is much more versatile than the FieldSet object of Parcels
2. by cutting the simulation into two simulations, for the first and second day separately. If you store the particle locations and then use [`ParticleSet.from_particlefile`](https://parcels.readthedocs.io/en/latest/#parcels.particleset.baseparticleset.BaseParticleSet.from_particlefile) to create a new FieldSet and ParticleSet with the ending locations of the first day simulation as starting locations
3. by writing your own Kernel that, based on the `time`, uses a different advection scheme (i.e. multiplying the velocities by a different scaling factor)
4. ... (there are probably many more ways to accomplish this)

Does that make sense?

**Reply by hblumberg** [reply]
Absolutely, thanks so much for confirming our thinking and for suggesting additional approaches!
